src/utils.jl

```diff
@@ -277,3 +277,83 @@ _eps_safe(::Type{T}) where {T<:Real} = eps(T)
 _eps_safe(::Type{T}) where {T<:Number} = eps(real(T))
 
 max_eps(xs...) = maximum(_eps_safe ∘ eltype, xs)
+
+"""
+    sectional_curvature_matrix(M::AbstractManifold, p, B::AbstractBasis)
+
+Compute the matrix of sectional curvatures of manifold `M` at point `p`.
+Entry `(i, j)` corresponds to sectional curvature of the surface spanned by vectors
+`i`  and `j` from basis `B`.
+"""
+function sectional_curvature_matrix(M::AbstractManifold, p, B::AbstractBasis)
+    V = get_vectors(M, p, get_basis(M, p, B))
+    N = length(V)
+    result = zeros(N, N)
+    for (i, e_i) in enumerate(V)
+        for (j, e_j) in enumerate(V)
+            if i < j
+                result[i, j] = sectional_curvature(M, p, e_i, e_j)
+                result[j, i] = result[i, j]
+            end
+        end
+    end
+    return result
+end
+
+@doc raw"""
+    sectional_curvature_bdp(M::AbstractManifold, p, X, Y; r::Real=1e-3, N::Int=10000)
+
+Approximate sectional curvature of manifold `M` in the plane spanned by vectors `X` and `Y`
+from tangent space at `p` using a circle on `M` of radius `r` divided into `N` segments.
+
+The approximation is derived from the [Bertrand–Diguet–Puiseux theorem](https://en.wikipedia.org/wiki/Bertrand%E2%80%93Diguet%E2%80%93Puiseux_theorem)
+which states that
+````math
+\kappa_p(X, Y) = \lim_{r \to 0^+} 3\frac{2\pi r-C(r)}{\pi r^3},
+````
+where ``C(r)`` is the circumference of the circle of radius ``r`` around `p` in submanifold
+of `M` spanned by `X` and `Y`. The circumference calculation method has a tendency to
+return curvature values larger than the exact ones.
+"""
+function sectional_curvature_bdp(M::AbstractManifold, p, X, Y; r::Real=1e-3, N::Int=10000)
```

Comment:
I had to check a while to see why ist is called `bdb`

maybe `estimated_sectional_curvature` (or approximated) would fit better? We could even use an `EstimationMethod` either now or later then (when there are multiple ones).

Reply:
I don't really want to over-engineer this in this PR, so maybe I will just rename it to `estimated_sectional_curvature`. I called it `_bdp` from the theorem name since there are other ways to estimate sectional curvature but we can handle that later.

Reply:
Yes, I read the doc string quite a few times until I saw where bpd is from.

I also think for now no method is necessary, just calling it `estimates_` should be fine.

Reply:
Updated :slightly_smiling_face: